knimin/lib/data_access.py

```diff
@@ -2032,6 +2053,32 @@ def search_handout_kits(self, term):
         liketerm = '%%' + term + '%%'
         return self._con.execute_fetchdict(sql, [liketerm, liketerm])
 
+    def get_handout_kits(self, kit_ids=None):
+        """Gets information on handout kits
+
+        Parameters
+        ----------
+        kit_ids : list of str, optional
+            List of kits to get info for. Default all handout kits
+
+        Returns
+        -------
+        list of dicts of objects
+            Each kit as a dict with all available info. Barcodes are a list
+            under the barcode key.
+        """
+        sql = """SELECT kit_id, verification_code, swabs_per_kit, created_on,
+                 array_agg(barcode) as barcodes
+                 FROM ag.ag_handout_kits
+                 JOIN ag.ag_handout_barcodes USING (kit_id)"""
+        sql_args = None
+        if kit_ids is not None:
+            sql += 'WHERE kit_ids IN %s'
```

Comment:
Missing a space before `WHERE`. Probably better to create `sql` as a list and then do a `' '.join(sql)` at the end. I think list appending is cheaper than string concatenation.